### PR TITLE
r2: add consistency & security docs

### LINKED ISSUE #7091: Highlight data consistency model for R2 S3 API
### Which Cloudflare product does this pertain to?

R2

### Existing documentation URL(s)

- https://developers.cloudflare.com/r2/data-access/s3-api/

### Section that requires update

Maybe: https://developers.cloudflare.com/r2/data-access/s3-api/api/

### What needs to change?

There was a question / uncertainty about the data consistency model of R2 on Discord. The current state of S3 API documentation does not reflect its strongly consistent nature. 

As AWS S3 was not strongly consistent [until 2020](https://aws.amazon.com/de/blogs/aws/amazon-s3-update-strong-read-after-write-consistency/) I can understand that someone can be uncertain about R2.

The only other place (I know of) which references it is the [beta announcement post](https://blog.cloudflare.com/r2-open-beta/) on the blog.

### How should it change?

The R2 S3 API documentation page should, like the [Workers API](https://developers.cloudflare.com/r2/data-access/workers-api/workers-api-reference/#bucket-method-definitions), reflect the data consistency model of R2.
I would suggest to add a new section to the S3 API Compatibility page which describes the model. 

### Additional information

Discord topic: 
- https://discord.com/channels/595317990191398933/940663374377783388/1035560497036148877
- https://discord.com/channels/595317990191398933/940663374377783388/1055431286380044289

### PR DESCRIPTION
**Pull request by elithrar**
Fixes #7091 

- A new "Data Security" page that covers how we encrypt data at rest, in-transit, and a link out to Cloudflare's security compliance certifications
- A new "Consistency Model" page that demonstrates R2's strong consistency on a per-operation basis

cc/ @dcpena

@codymgriffin @jonesphillip have co-reviewed internally.

### CODE REVIEW

File: content/r2/learning/consistency.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+This page details R2's consistency model, including where R2 is strongly, globally consistent and which operations this applies to.
+
+R2 can be described as "strongly consistent", especially in comparison to other distriubted object storage systems. This strong consistency ensures that operations against R2 see the latest (accurate) state: clients should be able to observe the effects of any write, update and/or delete operation immediately, globally. 
+
+## Terminology
+
+In the context of R2, _strong_ consistency and _eventual_ consistency have the following meanings: 
+
+* **Strongly consistent** - the effect of an operation will be observed globally, immediately, by all clients. Clients will not observe 'stale' (inconsistent) state.
```
Comment (dcpena) [inline content/r2/learning/consistency.md]:
```suggestion
* **Strongly consistent** - The effect of an operation will be observed globally, immediately, by all clients. Clients will not observe 'stale' (inconsistent) state.
```

File: content/r2/learning/consistency.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* **Eventually consistent** - clients may not see the effect of an operation immediately: state may take a some time (typically seconds to a minute) to propagate globally.
```
Comment (dcpena) [inline content/r2/learning/consistency.md]:
```suggestion
* **Eventually consistent** - Clients may not see the effect of an operation immediately. The state may take a some time (typically seconds to a minute) to propagate globally.
```

File: content/r2/learning/consistency.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+| Deletion: Delete an object                        | Strongly consistent: reads to that object will immediately return a "does not exist" error
+| Object listing: List the objects in a bucket      | Strongly consistent: the list operation will list all objects at that point in time                    |
+| IAM: Adding/removing R2 Storage permissions       | Eventually consistent: A [new or updated API key](https://developers.cloudflare.com/fundamentals/api/get-started/create-token/) may take up to a minute to have permissions reflected globally
+
+{{</table-wrap>}}
+
+Additional notes:
+
+* In the event two clients are writing (`PUT` or `DELETE`) to the same key, the last writer to complete "wins".
+* When performing a multipart upload, read-after-write consistency continues to apply once all parts have been successfully uploaded. In the case the same part is uploaded (in error) from multple writers, the last write will win
```
Comment (dcpena) [inline content/r2/learning/consistency.md]:
```suggestion
* When performing a multipart upload, read-after-write consistency continues to apply once all parts have been successfully uploaded. In the case the same part is uploaded (in error) from multiple writers, the last write will win.
```

File: content/r2/learning/consistency.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+By default, Cloudflare's cache will cache common, cacheable status codes automatically [per our cache documentation](https://developers.cloudflare.com/cache/how-to/configure-cache-status-code/#edge-ttl).
+
+{{</Aside>}}
+
+When connecting a [custom domain](https://developers.cloudflare.com/r2/data-access/public-buckets/#custom-domains) to an R2 bucket and enabling caching for objects served from that bucket, the consistency model is necessarily relaxed when accessing content via a domain with caching enabled.
+
+Specifically, you should expect:
+
+* An object you delete from R2, but that is still cached, will still be available. You should [purge the cache](https://developers.cloudflare.com/cache/how-to/purge-cache/) after deleting an object/objects if you need that delete to be reflected
```
Comment (dcpena) [inline content/r2/learning/consistency.md]:
```suggestion
* An object you delete from R2, but that is still cached, will still be available. You should [purge the cache](https://developers.cloudflare.com/cache/how-to/purge-cache/) after deleting objects if you need that delete to be reflected.
```

File: content/r2/learning/consistency.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+{{</Aside>}}
+
+When connecting a [custom domain](https://developers.cloudflare.com/r2/data-access/public-buckets/#custom-domains) to an R2 bucket and enabling caching for objects served from that bucket, the consistency model is necessarily relaxed when accessing content via a domain with caching enabled.
+
+Specifically, you should expect:
+
+* An object you delete from R2, but that is still cached, will still be available. You should [purge the cache](https://developers.cloudflare.com/cache/how-to/purge-cache/) after deleting an object/objects if you need that delete to be reflected 
+* Cloudflare's cache to [cache HTTP 404 (Not Found) responses](https://developers.cloudflare.com/cache/how-to/configure-cache-status-code/#edge-ttl) automatically. If you upload an object to that same path, the cache may continue to return HTTP 404s until the cache TTL (Time to Live) expires and the new object is fetched from R2, or the [cache is purged](https://developers.cloudflare.com/cache/how-to/purge-cache/).
```
Comment (dcpena) [inline content/r2/learning/consistency.md]:
```suggestion
* Cloudflare's cache to [cache HTTP 404 (Not Found) responses](https://developers.cloudflare.com/cache/how-to/configure-cache-status-code/#edge-ttl) automatically. If you upload an object to that same path, the cache may continue to return HTTP 404s until the cache TTL (Time to Live) expires and the new object is fetched from R2 or the [cache is purged](https://developers.cloudflare.com/cache/how-to/purge-cache/).
```

File: content/r2/learning/consistency.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+* An object for a given key is overwritten with a new object: the old (previous) object will continue to be served to clients until the cache TTL expires (or the object is evicted), or the cache is purged.
```
Comment (dcpena) [inline content/r2/learning/consistency.md]:
```suggestion
* An object for a given key is overwritten with a new object: the old (previous) object will continue to be served to clients until the cache TTL expires (or the object is evicted) or the cache is purged.
```

**Review by dcpena — COMMENTED** [review]
Few small fixes but otherwise 👍

**Review by elithrar — COMMENTED** [review]
Addresses changes @dcpena

**Review by dcpena — APPROVED** [review]

**Pull request merged**